### ISSUE TITLE
proxmox installation failed with docker

### ISSUE THREAD

**Issue by edouardsaucisse** [intuitem/ciso-assistant-community#314]
I followed the instruction to deploy a local instance of CISO Assistant, relying on docker.
I had some critical issues : 

- Installation seems buggy (non critical, just changed a command in the _docker-compose.sh_ script)
- Installation seems buggy n°2 : many issues in some python scripts (see error messages below)
- impossible to setup password (only mail address)
- impossible to connect to ssl page

**To Reproduce**
I follow the steps described in README.md, on the local installation section.
As I am in a ubuntu 22.04 with docker v24, I modified the _docker-compose.sh_ script to use _docker-compose_ command instead of _docker compose_. 
I do not expect it to have huge impacts on the following bugs.

**Expected behavior**
I expected the installation to go smooth as described

**Screenshots**
Screenshot of docker containers running : 
![image](https://github.com/intuitem/ciso-assistant-community/assets/167909047/76c0ec47-f682-4c61-8d04-1bae361de587)

Screenshot of http error
![image](https://github.com/intuitem/ciso-assistant-community/assets/167909047/38d415da-5db6-4dc7-ac74-4327564934bb)

Screenshot of ssl error
![image](https://github.com/intuitem/ciso-assistant-community/assets/167909047/8d33c574-0b83-4585-8b68-cabf38b7efd2)

**Environment (please complete the following information):**
_SERVER_
 - Device: virtual machine
 - OS: Ubuntu server 22.04, up-to-date
 - Docker : v24

_WORKSTATION_
- Windows 11 up-to-date
- Tried with chrome up-to-date
- Tried with firefox up-to-date

**Additional context**
when running _docker-compose.sh_ with _docker-compose_ commands instead of _docker compose_, i get these messages, detailling some python issues (dont consider the "Deleted" message at the beginning, they're just cleaning files from a previous try) : 

```shell

 root@ub-docker:~/ciso-assistant-community# ./docker-compose.sh -d
Untagged: ghcr.io/intuitem/ciso-assistant-community/backend:latest
Untagged: ghcr.io/intuitem/ciso-assistant-community/backend@sha256:386aaa2fddc454b5b8e78e9b22947f3a556619f7c17f92d3f376db4d545f4b91
Deleted: sha256:c50a8fffa08e777b5420ed13592b5f3155d3699b144f9e81705be1e20ac69c36
Deleted: sha256:1bdf9f50751163267085c451bb7371b32e7699980a8cec8e4ea366bba4d74dfb
Deleted: sha256:09c00f542ef83697ed6e131f34307606d7fe8e59f36677ee5283e685c8f69cad
Deleted: sha256:b8992b579269e2dcc480577ef3f27ed8b8f785ef589fbe4645733168ea66e777
Deleted: sha256:e0f3e4145488ca2f79531709a8ef139b859f328504b15af49475e6c84702fe44
Deleted: sha256:4153c2785d82d5e675a5df32f2b52ac20e1919b62a0a999a9762813b1246e9c6
Untagged: ghcr.io/intuitem/ciso-assistant-community/frontend:latest
Untagged: ghcr.io/intuitem/ciso-assistant-community/frontend@sha256:de8220c3b7836cb8e337518d76aa984c0a3272082acf7319b83cf0b70293451d
Deleted: sha256:a3baeecfde77858ab1c1f0d5c31766cf7f5b3553feae868d16a36bee6e580104
Deleted: sha256:29afb8b287bb5bc3ef9da5704d80134733bf28f1a5d93a5252d024b8e2c0af83
Deleted: sha256:8d1fc128364c8581930b6493343b552bea7c765e090974824519ac1ac0105ab8
Deleted: sha256:7a4ee7d27cb343b940e6c9acae4b3d0745b0ab7c6970425ac76eb1b2aee9597c
Deleted: sha256:ee6b37451864116861a6999a2892f952313036837ddfcfbda5eca08e64400c04
Deleted: sha256:b28846ef7ddedf16540419ff031f9093d2e8bd8ff6d420f478f5312733da5826
Pulling backend (ghcr.io/intuitem/ciso-assistant-community/backend:latest)...
latest: Pulling from intuitem/ciso-assistant-community/backend
609c73876867: Already exists
7247ea8d81e6: Already exists
be374d06f382: Already exists
b4580645a8e5: Already exists
aa7e0aca67dd: Already exists
928988b537f8: Already exists
95604f0c57a7: Already exists
af495b5540ac: Already exists
d78170b7f6d2: Pull complete
04e7329f7c85: Pull complete
e65e13aaae00: Pull complete
4f4fb700ef54: Pull complete
e97be9b3e9ca: Pull complete
Digest: sha256:386aaa2fddc454b5b8e78e9b22947f3a556619f7c17f92d3f376db4d545f4b91
Status: Downloaded newer image for ghcr.io/intuitem/ciso-assistant-community/backend:latest
Pulling frontend (ghcr.io/intuitem/ciso-assistant-community/frontend:latest)...
latest: Pulling from intuitem/ciso-assistant-community/frontend
4abcf2066143: Already exists
62af34686b6b: Already exists
87cba792d632: Already exists
67eb332999b7: Already exists
a255f4a49563: Pull complete
12cb41695cd7: Pull complete
0e773c7bbc87: Pull complete
6856bd2e9267: Pull complete
e055d02cde6c: Pull complete
Digest: sha256:de8220c3b7836cb8e337518d76aa984c0a3272082acf7319b83cf0b70293451d
Status: Downloaded newer image for ghcr.io/intuitem/ciso-assistant-community/frontend:latest
Creating backend ... done
Creating caddy    ... done
Creating frontend ... done
2024-04-23T17:52:15.208635Z [info     ] BASE_DIR: /code                [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:15.208826Z [info     ] VERSION: v1.1.1                [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:15.209406Z [info     ] BUILD: feb8456                 [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:15.209869Z [info     ] DEBUG mode: True               [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:15.210145Z [info     ] CISO_ASSISTANT_URL: https://localhost:8443 [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:15.210445Z [info     ] ALLOWED_HOSTS: ['backend']     [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:15.210851Z [info     ] SQLITE_FILE: /code/db/ciso-assistant.sqlite3 [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:15.211103Z [info     ] DATABASE ENGINE: django.db.backends.sqlite3 [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
Operations to perform:
  Apply all migrations: auth, cal, contenttypes, core, iam, knox, sessions
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0001_initial... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_alter_group_name_max_length... OK
  Applying auth.0011_update_proxy_permissions... OK
  Applying auth.0012_alter_user_first_name_max_length... OK
  Applying cal.0001_initial... OK
  Applying iam.0001_initial... OK
  Applying core.0001_initial... OK
  Applying core.0002_initial... OK
  Applying core.0003_alter_riskscenario_strength_of_knowledge... OK
  Applying core.0004_complianceassessment_is_published_and_more... OK
  Applying core.0005_alter_project_lc_status_alter_securitymeasure_effort... OK
  Applying core.0006_remove_securitymeasure_security_function_and_more...Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such table: core_securityfunction

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/code/manage.py", line 22, in <module>
    main()
  File "/code/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 413, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 459, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 107, in wrapper
    res = handle_func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/management/commands/migrate.py", line 356, in handle
    post_migrate_state = executor.migrate(
                         ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/migrations/executor.py", line 135, in migrate
    state = self._migrate_all_forwards(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards
    state = self.apply_migration(
            ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/migrations/executor.py", line 252, in apply_migration
    state = migration.apply(state, schema_editor)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/migrations/migration.py", line 132, in apply
    operation.database_forwards(
  File "/usr/local/lib/python3.11/site-packages/django/db/migrations/operations/models.py", line 440, in database_forwards
    schema_editor.alter_db_table(
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/schema.py", line 667, in alter_db_table
    self.execute(
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/schema.py", line 201, in execute
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "/usr/local/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such table: core_securityfunction
initialize your superuser account...
2024-04-23T17:52:21.036145Z [info     ] BASE_DIR: /code                [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:21.036529Z [info     ] VERSION: v1.1.1                [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:21.036858Z [info     ] BUILD: feb8456                 [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:21.037490Z [info     ] DEBUG mode: True               [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:21.037798Z [info     ] CISO_ASSISTANT_URL: https://localhost:8443 [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:21.038115Z [info     ] ALLOWED_HOSTS: ['backend']     [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:21.038603Z [info     ] SQLITE_FILE: /code/db/ciso-assistant.sqlite3 [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
2024-04-23T17:52:21.038842Z [info     ] DATABASE ENGINE: django.db.backends.sqlite3 [ciso_assistant.settings] ciso_assistant_url=https://localhost:8443
Email: bacon@eggs.com
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 134, in debug_sql
    yield
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'utf-8' codec can't encode character '\udcc2' in position 7: surrogates not allowed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/code/manage.py", line 22, in <module>
    main()
  File "/code/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 413, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.11/site-packages/django/contrib/auth/management/commands/createsuperuser.py", line 89, in execute
    return super().execute(*args, **options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 459, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/contrib/auth/management/commands/createsuperuser.py", line 131, in handle
    error_msg = self._validate_username(
                ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/contrib/auth/management/commands/createsuperuser.py", line 304, in _validate_username
    self.UserModel._default_manager.db_manager(database).get_by_natural_key(
  File "/usr/local/lib/python3.11/site-packages/django/contrib/auth/base_user.py", line 56, in get_by_natural_key
    return self.get(**{self.model.USERNAME_FIELD: username})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 645, in get
    num = len(clone)
          ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 382, in __len__
    self._fetch_all()
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 121, in execute
    with self.debug_sql(sql, params, use_last_executed_query=True):
  File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 139, in debug_sql
    sql = self.db.ops.last_executed_query(self.cursor, sql, params)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/sqlite3/operations.py", line 176, in last_executed_query
    params = self._quote_params_for_last_executed_query(params)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/sqlite3/operations.py", line 165, in _quote_params_for_last_executed_query
    return cursor.execute(sql, params).fetchone()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'utf-8' codec can't encode character '\udcc2' in position 7: surrogates not allowed
connect to ciso assistant on https://localhost:8443
for successive runs you can now use docker compose up

```

I guess it would deserve a split in several issues, but it's my first "contribution" on gh, so... forgive me if I'm doing wrong.

**Comment by ab-smith** [comment]
Hello, thank you for the feedback. Just to get it right, your setup is windows host with an ubuntu VM? which emulator or is it WSL? 

and you have attempted the docker-compose.sh script from within the VM or Windows command line?

We need indeed  to improve the instructions for Windows-based installation.

**Comment by edouardsaucisse** [comment]
Sorry, I realize my context was not clear :/
I use a dedicated server as a lab before production. It runs a proxmox hypervisor.
The hypervisor runs an ubuntu 22.04 VM, which embeds some docker virtual environments. 
Not the most efficient architecture, i know, but i feel it's close enough to production conditions and it allows me to tests stuff as I want.

The windows host is my desktop PC, which I use to manage proxmox (webUI) and connect to my virtual machines (inc. ubuntu 22.04 "docker" machine).

I hope it gets easier to understand and contextualize this way, but feel free to ask any precisions you want.

**Comment by Mohamed-Hacene** [comment]
Hi @edouardsaucisse, thanks for your contribution, it helps us a lot to better understand the different problems associated with different configurations.

I'll help you resolve these issues step by step.

Firstly, I strongly advise you to update your docker version if possible. `docker-compose` is deprecated for a year now (see [Migrate to Compose V2](https://docs.docker.com/compose/migrate/#docker-compose-vs-docker-compose)), although I don't think it's causing all these problems either, it's better to be up to date so that we can be sure it's not docker-related.

### Reproduction

I tried yesterday to install CISO Assistant on an Ubuntu server with the same version as you, with a remote access from my host machine but I didn't have the errors you encountered. So, let's check that everything is fine with your configuration before going any further with the code.

### Some checks:

1. The first thing I notice is that you're accessing your instance via `docker:8443` while the application is launched on `localhost:8443`. You need to modify the file `docker-compose.yml` to replace `localhost` occurrences by `docker` and add an argument in caddy service. I refer you to this tutorial: https://github.com/intuitem/ciso-assistant-community/issues/226#issuecomment-2051397358 which explains how to modify `docker-compose.yml` in your use case.
2. The first crash during docker-compose.sh script is related to the database. This may be due to failures in previous tests, which may have created conflicts between migrations and the database. Normally, the db folder at the root must be deleted to restart the docker-compose.sh script. If you have already done so, it would be better to restart from scratch.
3. If an error occurs during user creation, there is a simple workaround to create a user after the container creation: https://github.com/intuitem/ciso-assistant-community/issues/243#issuecomment-2051428147

### To summarize: 
- Update docker to compose V2 (strongly recommended)
- Restart from scratch the installation to be sure crashes come from previous failures and not from docker-compose.sh script
- Modify docker-compose.yml according to your configuration with the tutorial I sent

**Comment by edouardsaucisse** [comment]
Thanks for your answer !

I'll try these step on my current ubuntu VM.
If it fails, i'll try it out on a fresh and dedicated VM, to get rid of any environment related issue.

Just for information : if I host my CISOassistant instance on a remote server, addressing it by localhost:8443 will be pointless ^^
That's why I address it by docker:8443 : my ubuntu server is aliased "docker" in my DNS server (so I can reach it from anywhere on my network by simply calling http(s)://docker:whateverportorserviceIwant)

**Comment by Mohamed-Hacene** [comment]
Perfect !

Yes I understand, I know it comes from your DNS and it's the right things to do.

Without the FQDN, the SNI is not transmitted during the connection and caddy does not accept https without the SNI which causes an SSL error. What I meant was that yours is the other way around, where you've set the DNS correctly but the FQDN information you're using is missing from the docker-compose.yml, which also causes an SSL error. 

Sorry if my comment sounded presumptuous, that wasn't the point at all :)

**Comment by ab-smith** [comment]
@edouardsaucisse all good?

**Comment by ab-smith** [comment]
Hey, thanks for the feedback :) 
No, our focus is on all editions, and everyone is important. We have done multiple tests in different settings, but we might have missed some cases.
I'm currently setting up an equivalent enviornment to yours. If I can reproduce the issue we will fix it, otherwise, we can schedule a quick call through Discord to debug your specific case :)

**Comment by edouardsaucisse** [comment]
would be great !
Gonna join discord soon then :)

**Comment by ab-smith** [comment]
So good news, we are pretty confident that your error is originating from caddy settings for which you missed `-i` to generate the self-signed certificate for your `https://dockertmp:8443`  

We've tested on a lab like yours as well as a virtualization config with vagrant.
<img width="556" alt="image" src="https://github.com/intuitem/ciso-assistant-community/assets/11318384/87e42347-e1cc-4e82-b40d-3749952bb3e7">

and you'll get the ssl warning that you can accept (but not the blockage), and voilà :
<img width="708" alt="image" src="https://github.com/intuitem/ciso-assistant-community/assets/11318384/a8f11d32-527c-44d1-8759-f73890ad572b">

so, if you're starting fresh:
- make sure to clean the repo,
- do the three changes on the `docker-compose.yml`
- run `./docker-compose.sh`

and you should be good now 😊

for the migration random errors, could be related to a missing locale on your distro, we can talk it over later on.

Let me know

**Comment by ab-smith** [comment]
Added more explicit instructions here:

https://intuitem.gitbook.io/ciso-assistant/deployment/remote-virtualization

**Comment by LudovicARHIMAN** [comment]
> So good news, we are pretty confident that your error is originating from caddy with your lab settings for which you missed `-i` to generate the self-signed certificate for your `https://dockertmp:8443`
> 
> We've tested on a lab like yours as well as a virtualization config with vagrant. <img alt="image" width="556" src="https://private-user-images.githubusercontent.com/11318384/327606866-87e42347-e1cc-4e82-b40d-3749952bb3e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ4MjY0MzEsIm5iZiI6MTcxNDgyNjEzMSwicGF0aCI6Ii8xMTMxODM4NC8zMjc2MDY4NjYtODdlNDIzNDctZTFjYy00ZTgyLWI0MGQtMzc0OTk1MmJiM2U3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA0VDEyMzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0ZTA4NzYxODU4ZTFlMmFkYzRjOGIzZGQwZmJhMTA1ZDA0NTRhZmFiMDhiYjViOWZkNzc0NmZlNzliN2VmMzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UypoNbWfiI1OaF33cz5dD00YFRqXJV70-v4rEtDf4QQ">
> 
> and you'll get the ssl warning that you can accept (but not the blockage), and voilà : <img alt="image" width="708" src="https://private-user-images.githubusercontent.com/11318384/327607412-a8f11d32-527c-44d1-8759-f73890ad572b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ4MjY0MzEsIm5iZiI6MTcxNDgyNjEzMSwicGF0aCI6Ii8xMTMxODM4NC8zMjc2MDc0MTItYThmMTFkMzItNTI3Yy00NGQxLTg3NTktZjczODkwYWQ1NzJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA0VDEyMzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkZTIzMTRhMmMyMGM2NmZhM2U2ZGFhMTYxOWNlMTUyZTU1NTdjNjA1NWVhZGExNzJmOGEwZjc2NWUyMDg4YmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9pMIoyua59z4FcQOm1pX2Wl_Gm0Y4SkVjTneTmaO1cw">
> 
> so, if you're starting fresh:
> 
>     * make sure to clean the repo,
> 
>     * do the **three** changes on the `docker-compose.yml`
> 
>     * run `./docker-compose.sh`
> 
> 
> and you should be good now 😊
> 
> for the migration random errors, could be related to a missing locale on your distro, we can talk it over later on.
> 
> Given that proxmox is not the standard pattern for most users, we will probably add another flavor of docker-compose if you confirm that this fixes it. Keep in mind that the `-i` should not be used in production as it could present a security risk.
> 
> Let me know.


This fix works for me, I can see reach the page without any SSL problems.

**Comment by ab-smith** [comment]
@edouardsaucisse, all good?
